pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -913,6 +899,11 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 		}
 	}
 
+	var listenerFilters []listener.ListenerFilter
+	for _, filter := range listenerFiltersMap {
+		listenerFilters = append(listenerFilters, filter)
+	}
```

Comment:
This is not right. Map doesn’t guarantee iteration order. So the order of listener filters may be different from the one actually intended. Also each Envoy could get different permutations of listener filter chains.

Reply:
OK, now I know the order matters. Thanks.